src/Network/Transport/TCP/Internal.hs

```diff
@@ -183,44 +180,18 @@ forkServer host port backlog reuseAddr terminationHandler requestHandler = do
                                         (tryCloseSocket . fst)
                                         (requestHandler . fst)
 
--- | Read a length, then 1 or more payloads each less than some maximum
--- length in bytes, such that the sum of their lengths is the length that was
--- read.
-recvWithLengthFold
-  :: Word32                      -- ^ Maximum total size.
-  -> Word32                      -- ^ Maximum chunk size.
-  -> N.Socket
-  -> t                           -- ^ Start element for the fold.
-  -> ([ByteString] -> t -> IO t) -- ^ Run this every time we get data of at
-                                 -- most the maximum size.
-  -> IO t
-recvWithLengthFold maxSize maxChunk sock base folder = do
+-- | Read a length and then a payload of that length, subject to a limit
+--   on the length.
```

Comment:
Maybe worth note, what will happen if length is exceeded and where it's set. So reader will learn that without reading documentation in other places.